### PR TITLE
chainAction2

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] New `chainAction2` which infers typing based on the array of the action types. eventually we can swap out all chainAction, then rename chainAction2 to chainAction
- [x] Updated devices saga to use more async/await to demonstrate usage

With chainAction2 you can inline the call and not have to type state or the action but that might get messy. lets discuss in chat

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready. lets discuss

**Comment by chrisnojima** [comment]
- [ ] after release

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/actions/fs/platform-specific.desktop.tsx [hunk 1]
```diff
@@ -236,17 +234,21 @@ const fuseInstallResultIsKextPermissionError = (result: RPCTypes.InstallResult):
     c => c.name === 'fuse' && c.exitCode === Constants.ExitCodeFuseKextPermissionError
   ) !== -1
 
-const driverEnableFuse = (_, action: FsGen.DriverEnablePayload): Promise<Saga.MaybeAction> =>
-  RPCTypes.installInstallFuseRpcPromise().then((result: RPCTypes.InstallResult) =>
-    fuseInstallResultIsKextPermissionError(result)
-      ? [
+const driverEnableFuse = (_, action: FsGen.DriverEnablePayload): Saga.ChainActionReturn =>
+  new Promise(resolve => {
```
Comment (buoyad) [inline shared/actions/fs/platform-specific.desktop.tsx]: I don't think you need this `Promise`, removing it and adding `return`s gives no errors for me.

File: shared/actions/devices.tsx [hunk 2]
```diff
@@ -36,110 +40,114 @@ function* requestPaperKey(): Iterable<any> {
...
-  }
-
-  return RouteTreeGen.createNavigateAppend({
-    path: action.payload.wasCurrentDevice ? [Tabs.loginTab] : [...Constants.devicesTabLocation],
-  })
-}
+const navigateAfterRevoked = (_: Container.TypedState, action: DevicesGen.RevokedPayload) =>
+  action.payload.wasCurrentDevice
+    ? RouteTreeGen.createNavigateAppend({
+        path: action.payload.wasCurrentDevice ? [Tabs.loginTab] : [...Constants.devicesTabLocation],
```
Comment (buoyad) [inline shared/actions/devices.tsx]: I know not your change but this ternary isn't needed

File: shared/actions/fs/platform-specific.desktop.tsx (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/fs/platform-specific.desktop.tsx]: i converted it to async and cleaned that up, tnx!

**Pull request merged**